src/timesync/timesyncd-manager.c

```diff
@@ -1043,7 +1043,8 @@ static int manager_network_event_handler(sd_event_source *s, int fd, uint32_t re
         sd_network_monitor_flush(m->network_monitor);
 
         /* When manager_network_read_link_servers() failed, we assume that the servers are changed. */
-        changed = manager_network_read_link_servers(m);
+        r = manager_network_read_link_servers(m);
+        changed = r && r != -ENODATA;
```

Comment:
The change contradicts with the comment. Also, if we had some NTP servers, and now these are removed, then we could not detect the change...

Instead, please handle `-ENODATA` soon after `sd_network_get_ntp()`, e.g.
```diff
git diff
diff --git a/src/timesync/timesyncd-manager.c b/src/timesync/timesyncd-manager.c
index c8d38f212b..b8cc6f4ead 100644
--- a/src/timesync/timesyncd-manager.c
+++ b/src/timesync/timesyncd-manager.c
@@ -973,10 +973,10 @@ static int manager_network_read_link_servers(Manager *m) {
         assert(m);
 
         r = sd_network_get_ntp(&ntp);
-        if (r < 0) {
+        if (r < 0 && r != -ENODATA) {
                 if (r == -ENOMEM)
                         log_oom();
-                else if (r != -ENODATA)
+                else
                         log_debug_errno(r, "Failed to get link NTP servers: %m");
                 goto clear;
         }
```

Reply:
Thank you for the prompt response. 

> The change contradicts with the comment. Also, if we had some NTP servers, and now these are removed, then we could not detect the change...

This makes sense to me as well. I've just fixed and pushed the changes as you suggested and confirmed that the new commit still fixes the issue.